{
	"alias": "FEDORA-2024-9d2de2b051",
	"autotime": true,
	"bugs": [
		{
			"bug_id": 2255351,
			"feedback": [
				{
					"bug_id": 2255351,
					"comment": {
						"id": 3399126,
						"karma": 1,
						"timestamp": "2024-02-08 15:42:19",
						"update_id": 586412,
						"user": {
							"avatar": "https://seccdn.libravatar.org/avatar/2e1992bec165e697760cd5300381639a2a41273173ea22691431d8b5c776bca8?s=24&d=retro",
							"email": "guillaume.fieni@outlook.com",
							"groups": [
								{
									"name": "ipausers"
								},
								{
									"name": "signed_fpca"
								},
								{
									"name": "trust admins"
								}
							],
							"id": 7822,
							"name": "gfieni",
							"openid": "gfieni.id.fedoraproject.org"
						},
						"user_id": 7822
					},
					"comment_id": 3399126
				}
			],
			"security": true,
			"title": "CVE-2023-49083 python-cryptography: NULL-dereference when loading PKCS7 certificates [fedora-all]",
			"bugzilla": {
				"bug_id": "2255351",
				"creation_ts": "2023-12-20 10:45:40 +0000",
				"short_desc": "CVE-2023-49083 python-cryptography: NULL-dereference when loading PKCS7 certificates [fedora-all]",
				"delta_ts": "2024-03-21 08:39:36 +0000",
				"bug_status": "CLOSED",
				"resolution": "NOTABUG",
				"keywords": "Reopened, Security, SecurityTracking",
				"priority": "medium",
				"bug_severity": "medium",
				"blocked": [
					{
						"bug_id": "2255331",
						"alias": "CVE-2023-49083",
						"creation_ts": "2023-12-20 07:26:15 +0000",
						"short_desc": "CVE-2023-49083 python-cryptography: NULL-dereference when loading PKCS7 certificates",
						"delta_ts": "2024-12-11 18:24:36 +0000",
						"bug_status": "NEW",
						"keywords": "Security",
						"priority": "medium",
						"bug_severity": "medium",
						"depends_on": [
							"2255353",
							"2255358",
							"2255351",
							"2255352",
							"2255354",
							"2255356"
						],
						"blocked": [
							{
								"bug_id": "2255359",
								"error": "NotPermitted"
							}
						],
						"external_bugs": {
							"text": "RHSA-2024:3781",
							"name": "Red Hat Product Errata"
						},
						"long_desc": [
							{
								"isprivate": "0",
								"commentid": "17812104",
								"comment_count": "0",
								"who": {
									"text": "trathi",
									"name": "TEJ RATHI"
								},
								"bug_when": "2023-12-20 07:26:15 +0000",
								"thetext": "cryptography is a package designed to expose cryptographic primitives and recipes to Python developers. Calling `load_pem_pkcs7_certificates` or `load_der_pkcs7_certificates` could lead to a NULL-pointer dereference and segfault. Exploitation of this vulnerability poses a serious risk of Denial of Service (DoS) for any application attempting to deserialize a PKCS7 blob/certificate. The consequences extend to potential disruptions in system availability and stability. This vulnerability has been patched in version 41.0.6.\n\nhttps://github.com/pyca/cryptography/commit/f09c261ca10a31fe41b1262306db7f8f1da0e48a\nhttps://github.com/pyca/cryptography/pull/9926\nhttps://github.com/pyca/cryptography/security/advisories/GHSA-jfhm-5ghh-2f97"
							},
							{
								"isprivate": "0",
								"commentid": "17812280",
								"comment_count": "1",
								"who": {
									"text": "trathi",
									"name": "TEJ RATHI"
								},
								"bug_when": "2023-12-20 10:46:13 +0000",
								"thetext": "Created python-cryptography tracking bugs for this issue:\n\nAffects: fedora-all [bug 2255351]\nAffects: openstack-rdo [bug 2255352]\n\n\nCreated python3-cryptography tracking bugs for this issue:\n\nAffects: epel-all [bug 2255353]"
							},
							{
								"isprivate": "0",
								"commentid": "17891831",
								"comment_count": "8",
								"who": {
									"text": "updates",
									"name": "Fedora Update System"
								},
								"bug_when": "2024-02-17 00:56:21 +0000",
								"thetext": "FEDORA-2024-91f5df4002 (python-cryptography-41.0.7-1.fc39) has been pushed to the Fedora 39 stable repository.\nIf problem still persists, please make note of it in this bug report."
							},
							{
								"isprivate": "0",
								"commentid": "17955050",
								"comment_count": "10",
								"who": {
									"text": "errata-xmlrpc",
									"name": "errata-xmlrpc"
								},
								"bug_when": "2024-04-02 19:30:14 +0000",
								"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Ansible Automation Platform 2.4 for RHEL 9\n  Red Hat Ansible Automation Platform 2.4 for RHEL 8\n\nVia RHSA-2024:1640 https://access.redhat.com/errata/RHSA-2024:1640"
							},
							{
								"isprivate": "0",
								"commentid": "17976676",
								"comment_count": "11",
								"who": {
									"text": "errata-xmlrpc",
									"name": "errata-xmlrpc"
								},
								"bug_when": "2024-04-18 01:52:01 +0000",
								"thetext": "This issue has been addressed in the following products:\n\n  RHUI 4 for RHEL 8\n\nVia RHSA-2024:1878 https://access.redhat.com/errata/RHSA-2024:1878"
							},
							{
								"isprivate": "0",
								"commentid": "17993997",
								"comment_count": "12",
								"who": {
									"text": "errata-xmlrpc",
									"name": "errata-xmlrpc"
								},
								"bug_when": "2024-04-30 10:04:12 +0000",
								"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Enterprise Linux 9\n\nVia RHSA-2024:2337 https://access.redhat.com/errata/RHSA-2024:2337"
							},
							{
								"isprivate": "0",
								"commentid": "18027478",
								"comment_count": "13",
								"who": {
									"text": "errata-xmlrpc",
									"name": "errata-xmlrpc"
								},
								"bug_when": "2024-05-22 09:45:35 +0000",
								"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Enterprise Linux 8\n\nVia RHSA-2024:3105 https://access.redhat.com/errata/RHSA-2024:3105"
							},
							{
								"isprivate": "0",
								"commentid": "18060134",
								"comment_count": "14",
								"who": {
									"text": "errata-xmlrpc",
									"name": "errata-xmlrpc"
								},
								"bug_when": "2024-06-10 18:36:58 +0000",
								"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Ansible Automation Platform 2.4 for RHEL 9\n  Red Hat Ansible Automation Platform 2.4 for RHEL 8\n\nVia RHSA-2024:3781 https://access.redhat.com/errata/RHSA-2024:3781"
							},
							{
								"isprivate": "0",
								"commentid": "18299190",
								"comment_count": "16",
								"who": {
									"text": "errata-xmlrpc",
									"name": "errata-xmlrpc"
								},
								"bug_when": "2024-12-11 18:24:32 +0000",
								"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Enterprise Linux 8.8 Extended Update Support\n\nVia RHSA-2024:10965 https://access.redhat.com/errata/RHSA-2024:10965"
							}
						]
					}
				],
				"external_bugs": {
					"text": "FREEIPA-10685",
					"name": "Red Hat Issue Tracker"
				},
				"long_desc": [
					{
						"isprivate": "0",
						"commentid": "17812275",
						"comment_count": "0",
						"who": {
							"text": "trathi",
							"name": "TEJ RATHI"
						},
						"bug_when": "2023-12-20 10:45:40 +0000",
						"thetext": "\nMore information about this security flaw is available in the following bug:\n\nhttp://bugzilla.redhat.com/show_bug.cgi?id=2255331\n\nDisclaimer: Community trackers are created by Red Hat Product Security team on a best effort basis. Package maintainers are required to ascertain if the flaw indeed affects their package, before starting the update process."
					},
					{
						"isprivate": "0",
						"commentid": "17812276",
						"comment_count": "1",
						"who": {
							"text": "trathi",
							"name": "TEJ RATHI"
						},
						"bug_when": "2023-12-20 10:45:44 +0000",
						"thetext": "Use the following template to for the 'fedpkg update' request to submit an\nupdate for this issue as it contains the top-level parent bug(s) as well as\nthis tracking bug.  This will ensure that all associated bugs get updated\nwhen new packages are pushed to stable.\n\n=====\n\n# bugfix, security, enhancement, newpackage (required)\ntype=security\n\n# low, medium, high, urgent (required)\nseverity=medium\n\n# testing, stable\nrequest=testing\n\n# Bug numbers: 1234,9876\nbugs=2255331,2255351\n\n# Description of your update\nnotes=Security fix for [PUT CVEs HERE]\n\n# Enable request automation based on the stable/unstable karma thresholds\nautokarma=True\nstable_karma=3\nunstable_karma=-3\n\n# Automatically close bugs when this marked as stable\nclose_bugs=True\n\n# Suggest that users restart after update\nsuggest_reboot=False\n\n======\n\nAdditionally, you may opt to use the bodhi web interface to submit updates:\n\nhttps://bodhi.fedoraproject.org/updates/new"
					},
					{
						"isprivate": "0",
						"commentid": "17871127",
						"comment_count": "2",
						"who": {
							"text": "code",
							"name": "Ben Beasley"
						},
						"bug_when": "2024-02-01 15:43:09 +0000",
						"thetext": "This would be fixed by updating to at least 41.0.6. (41.0.7 is the latest 41.0.x release.)"
					},
					{
						"isprivate": "0",
						"commentid": "17874049",
						"comment_count": "3",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-02-04 16:02:57 +0000",
						"thetext": "FEDORA-2024-9d2de2b051 has been submitted as an update to Fedora 40. https://bodhi.fedoraproject.org/updates/FEDORA-2024-9d2de2b051"
					},
					{
						"isprivate": "0",
						"commentid": "17874133",
						"comment_count": "4",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-02-04 17:48:23 +0000",
						"thetext": "FEDORA-2024-9d2de2b051 has been pushed to the Fedora 40 stable repository.\nIf problem still persists, please make note of it in this bug report."
					},
					{
						"isprivate": "0",
						"commentid": "17874160",
						"comment_count": "5",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-02-04 18:55:06 +0000",
						"thetext": "FEDORA-2024-91f5df4002 has been submitted as an update to Fedora 39. https://bodhi.fedoraproject.org/updates/FEDORA-2024-91f5df4002"
					},
					{
						"isprivate": "0",
						"commentid": "17874457",
						"comment_count": "6",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-02-05 01:12:06 +0000",
						"thetext": "FEDORA-2024-91f5df4002 has been pushed to the Fedora 39 testing repository.\nSoon you'll be able to install the update with the following command:\n`sudo dnf upgrade --enablerepo=updates-testing --refresh --advisory=FEDORA-2024-91f5df4002`\nYou can provide feedback for this update here: https://bodhi.fedoraproject.org/updates/FEDORA-2024-91f5df4002\n\nSee also https://fedoraproject.org/wiki/QA:Updates_Testing for more information on how to test updates."
					},
					{
						"isprivate": "0",
						"commentid": "17891832",
						"comment_count": "7",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-02-17 00:56:25 +0000",
						"thetext": "FEDORA-2024-91f5df4002 (python-cryptography-41.0.7-1.fc39) has been pushed to the Fedora 39 stable repository.\nIf problem still persists, please make note of it in this bug report."
					},
					{
						"isprivate": "0",
						"commentid": "17937832",
						"comment_count": "8",
						"who": {
							"text": "cheimes",
							"name": "Christian Heimes"
						},
						"bug_when": "2024-03-21 08:39:36 +0000",
						"thetext": "Fixed in F39 and newer."
					}
				]
			}
		}
	],
	"builds": [
		{
			"epoch": 0,
			"nvr": "python-cryptography-41.0.7-1.fc40",
			"release_id": 76,
			"signed": true,
			"type": "rpm",
			"package": {
				"aarch64": [
					{
						"name": "python-cryptography-debugsource",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "aarch64"
					},
					{
						"name": "python3-cryptography-debuginfo",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "aarch64"
					},
					{
						"name": "python3-cryptography",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "aarch64"
					}
				],
				"i686": [
					{
						"name": "python3-cryptography-debuginfo",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "i686"
					},
					{
						"name": "python-cryptography-debugsource",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "i686"
					},
					{
						"name": "python3-cryptography",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "i686"
					}
				],
				"ppc64le": [
					{
						"name": "python-cryptography-debugsource",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "ppc64le"
					},
					{
						"name": "python3-cryptography-debuginfo",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "ppc64le"
					},
					{
						"name": "python3-cryptography",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "ppc64le"
					}
				],
				"s390x": [
					{
						"name": "python3-cryptography-debuginfo",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "s390x"
					},
					{
						"name": "python-cryptography-debugsource",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "s390x"
					},
					{
						"name": "python3-cryptography",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "s390x"
					}
				],
				"src": [
					{
						"name": "python-cryptography",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "src"
					}
				],
				"x86_64": [
					{
						"name": "python3-cryptography-debuginfo",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "x86_64"
					},
					{
						"name": "python-cryptography-debugsource",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "x86_64"
					},
					{
						"name": "python3-cryptography",
						"version": "41.0.7",
						"release": "1.fc40",
						"arch": "x86_64"
					}
				]
			}
		}
	],
	"close_bugs": true,
	"comments": [
		{
			"id": 3391842,
			"text": "This update was automatically created",
			"timestamp": "2024-02-04 16:02:54",
			"update_id": 586369,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 3391843,
			"text": "This update's test gating status has been changed to 'waiting'.",
			"timestamp": "2024-02-04 16:03:07",
			"update_id": 586369,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 3391928,
			"text": "This update's test gating status has been changed to 'passed'.",
			"timestamp": "2024-02-04 17:45:43",
			"update_id": 586369,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 3391929,
			"text": "This update has been submitted for stable by bodhi",
			"timestamp": "2024-02-04 17:48:20",
			"update_id": 586369,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		}
	],
	"content_type": "rpm",
	"critpath": true,
	"critpath_groups": "critical-path-compose critical-path-server",
	"date_approved": "2024-02-04 17:48:20",
	"date_pushed": "2024-02-04 17:48:20",
	"date_stable": "2024-02-04 17:48:20",
	"date_submitted": "2024-02-04 16:02:54",
	"date_testing": "2024-02-04 16:03:07",
	"meets_testing_requirements": true,
	"notes": "Automatic update for python-cryptography-41.0.7-1.fc40.\n\n##### **Changelog**\n\n```\n* Thu Feb  1 2024 Benjamin A. Beasley <code@musicinmybrain.net> - 41.0.7-1\n- Update to 41.0.7, fixes rhbz#2255351, CVE-2023-49083\n\n```",
	"pushed": true,
	"release": {
		"branch": "f40",
		"candidate_tag": "f40-updates-candidate",
		"composed_by_bodhi": true,
		"create_automatic_updates": false,
		"dist_tag": "f40",
		"eol": "2025-05-13",
		"id_prefix": "FEDORA",
		"long_name": "Fedora 40",
		"mail_template": "fedora_errata_template",
		"name": "F40",
		"override_tag": "f40-override",
		"package_manager": "dnf",
		"pending_signing_tag": "f40-signing-pending",
		"pending_stable_tag": "f40-updates-pending",
		"pending_testing_tag": "f40-updates-testing-pending",
		"stable_tag": "f40-updates",
		"state": "current",
		"testing_repository": "updates-testing",
		"testing_tag": "f40-updates-testing",
		"version": "40"
	},
	"severity": "unspecified",
	"stable_karma": 3,
	"status": "stable",
	"suggest": "unspecified",
	"test_gating_status": "passed",
	"title": "python-cryptography-41.0.7-1.fc40",
	"type": "security",
	"url": "https://bodhi.fedoraproject.org/updates/FEDORA-2024-9d2de2b051",
	"unstable_karma": -3,
	"updateid": "FEDORA-2024-9d2de2b051",
	"user": {
		"avatar": "https://seccdn.libravatar.org/avatar/e221b70024cf971d8b43de0260d2babde9179cdf72605bf3168420e3efe07423?s=24&d=retro",
		"email": "mhroncok@redhat.com",
		"groups": [
			{
				"name": "provenpackager"
			},
			{
				"name": "packager"
			},
			{
				"name": "python-sig"
			},
			{
				"name": "signed_fpca"
			},
			{
				"name": "3d-printing-sig"
			},
			{
				"name": "python-packagers-sig"
			},
			{
				"name": "fedora-contributor"
			},
			{
				"name": "sysadmin-badges"
			},
			{
				"name": "ipausers"
			},
			{
				"name": "sysadmin"
			},
			{
				"name": "lua-packagers-sig"
			},
			{
				"name": "fedorabugs"
			},
			{
				"name": "cvsadmin"
			},
			{
				"name": "ambassadors"
			},
			{
				"name": "packaging-committee"
			},
			{
				"name": "advocates"
			}
		],
		"id": 533,
		"name": "churchyard",
		"openid": "churchyard.id.fedoraproject.org"
	},
	"version_hash": "db06a4a304134045c6022d1ab271bc7c4a0471ae"
}
